admin/src/main/java/org/bahmni/module/admin/csv/persister/LabResultPersister.java

```diff
@@ -133,10 +128,18 @@ private Order getTestOrder(Patient patient, LabResultRow labResultRow, Date test
         return order;
     }
 
-    private Obs getResultObs(LabResultRow labResultRow, Order testOrder) {
+    private Obs getResultObs(LabResultRow labResultRow, Order testOrder, Concept concept) {
         LabOrderResult labOrderResult = new LabOrderResult();
-        labOrderResult.setResult(labResultRow.getResult());
+        String result = labResultRow.getResult();
+        labOrderResult.setResult(result);
         labOrderResult.setResultDateTime(testOrder.getDateActivated());
+        if (concept.getDatatype().getHl7Abbreviation().equals("CWE")) {
+            Concept resultConcept = conceptService.getConceptByName(result);
```

Comment:
conceptService.getConceptByName(result)?? 
or should it be 
conceptService.getConceptByName(testOrder.getConcept())??

Reply:
For **CODED** answer `result` is a concept. So, to map it properly you have to get the concept for the result.

Reply:
thats right.